numpy/lib/_datasource.py

```diff
@@ -333,12 +333,9 @@ def _cache(self, path):
 
         # TODO: Doesn't handle compressed files!
         if self._isurl(path):
-            try:
-                with closing(urlopen(path)) as openedurl:
-                    with _open(upath, 'wb') as f:
-                        shutil.copyfileobj(openedurl, f)
-            except URLError:
-                raise URLError("URL not found: %s" % path)
+            with closing(urlopen(path)) as openedurl:
```

Comment:
I bet `closing` isn't needed any more either...

Reply:
Yes, making another commit.
Thank you